### PR TITLE
In-place predict for higgs

### PR DESCRIPTION
**Pull request by ShvetsKS**

### CODE REVIEW

**Comment by ShvetsKS** [comment]
@Alexsandruss @PetrovKP could you take a look at updated xgboost benchmark?

File: xgboost/gbt.py [hunk 1]
```diff
@@ -67,6 +67,8 @@ def convert_xgb_predictions(y_pred, objective):
                     help='Control a balance of positive and negative weights')
 parser.add_argument('--count-dmatrix', default=False, action='store_true',
                     help='Count DMatrix creation in time measurements')
+parser.add_argument('--inplace-predict', default=False, action='store_true',
+                    help='Count DMatrix creation in time measurements')
```
Comment (Alexsandruss) [inline xgboost/gbt.py:71]: Copy-pasted help from previous argument, write own

File: configs/xgb_cpu_config.json [hunk 2]
```diff
@@ -131,7 +131,8 @@
             "n-estimators": [1000],
             "objective": ["binary:logistic"],
             "tree-method": ["hist"],
-            "enable-experimental-json-serialization": ["False"]
+            "enable-experimental-json-serialization": ["False"],
+            "inplace-predict": [""]
```
Comment (Alexsandruss) [inline configs/xgb_cpu_config.json:135]: Are same changes needed for XGB GPU config?

File: xgboost/gbt.py [hunk 3]
```diff
@@ -135,9 +137,13 @@ def fit():
...
-    def predict():
-        dtest = xgb.DMatrix(X_test, y_test)
-        return booster.predict(dtest)
+    if params.inplace_predict == False:
+        def predict():
+            dtest = xgb.DMatrix(X_test, y_test)
+            return booster.predict(dtest)
+    else:
+        def predict():
+            return booster.inplace_predict(np.ascontiguousarray(X_test.values, dtype=np.float32))
```
Comment (Alexsandruss) [inline xgboost/gbt.py:146]: Why np.ascontiguousarray?

**Review by Alexsandruss — CHANGES_REQUESTED** [review]

File: xgboost/gbt.py (hunk 3 above)
Comment (ShvetsKS) [reply xgboost/gbt.py:146]: inplace supports only numpy c_contigous array

File: configs/xgb_cpu_config.json (hunk 2 above)
Comment (ShvetsKS) [reply configs/xgb_cpu_config.json:135]: done

File: xgboost/gbt.py (hunk 1 above)
Comment (ShvetsKS) [reply xgboost/gbt.py]: done

File: xgboost/gbt.py (hunk 3 above)
Comment (ShvetsKS) [reply xgboost/gbt.py:146]: I hope Sklearn API will support this method soon for pandas and numpy.

Comment (Alexsandruss) [reply xgboost/gbt.py:146]: And float32 only?

Comment (ShvetsKS) [reply xgboost/gbt.py:146]: If data is `float64` same conversion will be performed inside `inplace_predict`
So as a conversion is included to measurement I don't think that there is any problem

**Review by Alexsandruss — APPROVED** [review]

**Pull request merged**